core/jvm/src/test/scala/scalaz/zio/internal/RingBufferSpec.scala

```diff
@@ -41,4 +43,11 @@ class RingBufferSpec extends Specification {
       .and(q.poll(-1) must_=== -1)
       .and(q.isEmpty must beTrue)
   }
+
+  def e4 = {
+    val q = MutableConcurrentQueue.bounded[Int](2)
+    q.offer(1)
+    val returnQ = serializeAndDeserialize(q)
+    returnQ.poll(-1) must_=== 1
```

Comment:
Add `returnQ.offer(2)` right here before the poll and you will reproduce the infinite loop.

Reply:
@ghostdogpr ah, interesting! Thanks for pointing out 👍